src/Microsoft.IdentityModel.Tokens/CompressionProviderFactory.cs

```diff
@@ -0,0 +1,119 @@
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+using System;
+using Microsoft.IdentityModel.Logging;
+
+namespace Microsoft.IdentityModel.Tokens
+{
+    /// <summary>
+    /// Compression provider factory for compression and decompression.
+    /// </summary>
+    public class CompressionProviderFactory
+    {
+        private static CompressionProviderFactory _default;
+
+        /// <summary>
+        /// Static constructor that initializes the default <see cref="CompressionProviderFactory"/>.
+        /// </summary>
+        static CompressionProviderFactory()
+        {
+            Default = new CompressionProviderFactory();
+        }
+
+        /// <summary>
+        /// Default constructor for <see cref="CompressionProviderFactory"/>.
+        /// </summary>
+        public CompressionProviderFactory()
+        {
+        }
+
+        /// <summary>
+        /// Constructor that creates a deep copy of given <see cref="CompressionProviderFactory"/> object.
+        /// </summary>
+        /// <param name="other"><see cref="CompressionProviderFactory"/> to copy from.</param>
+        public CompressionProviderFactory(CompressionProviderFactory other)
+        {
+            if (other == null)
+                throw LogHelper.LogArgumentNullException(nameof(other));
+
+            CustomCompressionProvider = other.CustomCompressionProvider;
+        }
+
+        /// <summary>
+        /// Returns the default <see cref="CompressionProviderFactory"/> instance.
+        /// </summary>
+        public static CompressionProviderFactory Default
+        {
+            get => _default;
+            set => _default = value ?? throw LogHelper.LogArgumentNullException(nameof(Default));
+        }
+
+        /// <summary>
+        /// Extensibility point for custom compression support application wide.
+        /// </summary>
+        public CompressionProvider CustomCompressionProvider { get; set; }
```

Comment:
If you'd like this to be extensible, then it would be good to create an ICompressionProvider interface and declare this property as that type instead.

Reply:
Done. I created an ICompressionProvider interface.